platform-sdk/swirlds-common/src/main/java/com/swirlds/common/config/ConfigUtils.java

```diff
@@ -33,62 +38,78 @@ public final class ConfigUtils {
-            return configurationBuilder;
+    @NonNull
+    public static ConfigurationBuilder scanAndRegisterAllConfigTypes(
+            @NonNull final ConfigurationBuilder configurationBuilder, @Nullable final String... packagePrefixes) {
+        CommonUtils.throwArgNull(configurationBuilder, "configurationBuilder");
+        if (packagePrefixes == null) {
+            return scanAndRegisterAllConfigTypes(configurationBuilder, Collections.emptySet(), Collections.emptyList());
+        } else {
+            return scanAndRegisterAllConfigTypes(
+                    configurationBuilder, Set.of(packagePrefixes), Collections.emptyList());
         }
     }
 
     /**
      * Scan all classes in a classpath and register all configuration data types with a configuration builder.
      *
-     * @param configurationBuilder
-     * 		a configuration builder
-     * @param packagePrefix
-     * 		the package prefix to scan
-     * @param additionalClassLoaders
-     * 		additional classloaders to scan
+     * @param configurationBuilder   a configuration builder
+     * @param packagePrefixes        the package prefixes to scan
+     * @param additionalClassLoaders additional classloaders to scan
+     * @return the configuration builder that was passed as a param (for fluent api)
      */
-    @SuppressWarnings("unchecked")
-    public static void scanAndRegisterAllConfigTypes(
-            final ConfigurationBuilder configurationBuilder,
-            final String packagePrefix,
-            final URLClassLoaderWithLookup... additionalClassLoaders) {
+    @NonNull
+    public static ConfigurationBuilder scanAndRegisterAllConfigTypes(
+            @NonNull final ConfigurationBuilder configurationBuilder,
+            @NonNull final Set<String> packagePrefixes,
+            @NonNull final List<URLClassLoaderWithLookup> additionalClassLoaders) {
+        loadAllConfigDataRecords(packagePrefixes, additionalClassLoaders)
+                .forEach(configurationBuilder::withConfigDataType);
+        return configurationBuilder;
+    }
 
-        final ClassGraph classGraph = new ClassGraph().enableAnnotationInfo().whitelistPackages(packagePrefix);
+    @NonNull
+    private static Set<Class<? extends Record>> loadAllConfigDataRecords(
+            @NonNull final Set<String> packagePrefixes,
+            @NonNull final List<URLClassLoaderWithLookup> additionalClassLoaders) {
+        CommonUtils.throwArgNull(packagePrefixes, "packagePrefix");
+        CommonUtils.throwArgNull(additionalClassLoaders, "additionalClassLoaders");
 
-        if (additionalClassLoaders != null) {
+        final ClassGraph classGraph = new ClassGraph().enableAnnotationInfo();
+
+        if (!packagePrefixes.isEmpty()) {
+            classGraph.whitelistPackages(packagePrefixes.toArray(new String[packagePrefixes.size()]));
+        }
+
+        if (!additionalClassLoaders.isEmpty()) {
```

Comment:
This check is not needed

Reply:
classGraph.whitelistPackages() set some flags before iterating over the array. Based on this I do not know if we can simply remove the check.

Reply:
Not sure I understand you. `additionClassLoaders` is a simple `List`. If it is empty, iterating over it should be a no-op.

Reply:
lol, you are right. My fault. I looked at like 98 instead of 102 ;)